java/code/src/com/suse/oval/OvalParser.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.suse.oval;
+
+import com.suse.oval.exceptions.OvalParserException;
+import com.suse.oval.ovaltypes.OvalRootType;
+
+import java.io.File;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+
+/**
+ * The Oval Parser is responsible for parsing OVAL(Open Vulnerability and Assessment Language) documents
+ */
+public class OvalParser {
+
+    /**
+     * Parse the given OVAL file
+     *
+     * @param ovalFile the OVAL file to parse
+     * @return the parsed OVAL encapulated in a {@link OvalRootType} object=
+     * */
+    public OvalRootType parse(File ovalFile) throws OvalParserException {
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(OvalRootType.class);
```

Comment:
I think it should be using "StAX" instead of Jaxb, at least according to the issue: https://github.com/uyuni-project/uyuni/issues/7108

Also, I'm probably missing something but where is this class used outside tests?

Reply:
I think I found the reason for not being called outside the tests. Is it because the download and automatic extract are not yet implemented?